altapay/altapay.php

```diff
@@ -2070,156 +2077,205 @@ private function decodeHtmlEntitiesArrayValues($arr)
         return $arr;
     }
 
-    //function will be used to create the capture or refund quantity count in order to store in the db
-
-    /** @var CartCore */
+    /**
+     * function will be used to create the capture or refund quantity count in order to store in the db
+     * @return array
+     * @var CartCore
+     */
     private function getOrderLines($cart)
```

Comment:
Changes regarding the free gift, cart rules with amount are made in this method. This method has been refactored into small methods and different scenarios are being tackled accordingly